pyiso/nyiso.py

```diff
@@ -157,8 +158,9 @@ def get_any(self, label, parser, dates_list=None):
 
             # if fetch_csvs cannot get the individual days, it gets the whole month
             # Shortcut the loop if any call to fetch_csvs gets all dates in dates_list
+            date = pieces[-1].index[-1]
             try:
-                if (pieces[-1].index[-1] - timedelta(days=1)).date() > max(dates_list):
+                if (date - timedelta(days=1)).date() > max(dates_list):
```

Comment:
Given you're able to fix the `datetime` issue on 161, there's a second issue.

For a `dates_list` with dates too far into the future that `fetch_csvs(...)` will never increment and will always be less than the maximum date in `dates_list` and this method gets stuck in an infinite loop.